From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MrIvanPlays <ivan@mrivanplays.com>
Date: Mon, 27 Jul 2020 14:11:29 +0300
Subject: [PATCH] Revert Paper 7db30eb

Crashes started coming from everywhere after Aikar introduced this change in order to fix a lightning bug.
This should (theoretically) fix the crashes.
Yes if this fix works we're basically reverting a (half) working fix of a weird lightning bug, but as they say - in order to fix 1, you create 2.

diff --git a/src/main/java/net/minecraft/server/LightEngineThreaded.java b/src/main/java/net/minecraft/server/LightEngineThreaded.java
index 721f86a5405c9575d8ee3fc661395e65487b6d96..30c3997761c63b64ca2054c62633ffbe077e1e6f 100644
--- a/src/main/java/net/minecraft/server/LightEngineThreaded.java
+++ b/src/main/java/net/minecraft/server/LightEngineThreaded.java
@@ -230,7 +230,10 @@ public class LightEngineThreaded extends LightEngine implements AutoCloseable {
         // Paper start
         long pair = chunkcoordintpair.pair();
         CompletableFuture<IChunkAccess> future = new CompletableFuture<>();
-        IntSupplier prioritySupplier = d.getPrioritySupplier(pair);
+        // Origami start
+        IntSupplier prioritySupplier1 = d.getPrioritySupplier(pair);
+        IntSupplier prioritySupplier = flag ? () -> Math.max(1, prioritySupplier1.getAsInt() - 10) : prioritySupplier1;
+        // Origami end
         this.e.a(ChunkTaskQueueSorter.a(() -> {
             this.queue.add(pair, prioritySupplier.getAsInt(), LightEngineThreaded.Update.PRE_UPDATE, SystemUtils.a(() -> {
                 // Paper end
